src/transactions/test/PathPaymentStrictSendTests.cpp

```diff
@@ -2434,3 +2332,132 @@ TEST_CASE("pathpayment strict send", "[tx][pathpayment]")
         }
     }
 }
+
+TEST_CASE("pathpayment strict send uses all offers in a loop",
+          "[tx][pathpayment]")
+{
+    // This test would downgrade the bucket protocol from >12 to 12
+    // with USE_CONFIG_FOR_GENESIS.  Some other tests in this module,
+    // however, rely on that being set, so we separate this one
+    // out into a test case with its own Application object.
+    Config cfg = getTestConfig();
+    cfg.USE_CONFIG_FOR_GENESIS = false;
+    VirtualClock clock;
+    auto app = createTestApplication(clock, cfg);
+    app->start();
+    auto& lm = app->getLedgerManager();
+    auto const txfee = lm.getLastTxFee();
+    auto const minBalance1 = lm.getLastMinBalance(1) + 10 * txfee;
+    auto const minBalance3 = lm.getLastMinBalance(3) + 10 * txfee;
+    auto const minBalance4 = lm.getLastMinBalance(4) + 10 * txfee;
+    auto const minBalance5 = lm.getLastMinBalance(5) + 10 * txfee;
+    auto root = TestAccount::createRoot(*app);
+    auto gateway = root.create("gate1", minBalance5);
+    auto gateway2 = root.create("gate2", minBalance5);
+
+    auto useAllOffersInLoop = [&](TestAccount* issuerToDelete) {
+        for_versions_from(12, *app, [&] {
+            auto market = TestMarket{*app};
+            auto source = root.create("source", minBalance4);
+            auto destination = root.create("destination", minBalance1);
+            auto mm12 = root.create("mm12", minBalance3);
+            auto mm23 = root.create("mm23", minBalance3);
+            auto mm34 = root.create("mm34", minBalance3);
+            auto mm41 = root.create("mm41", minBalance3);
+            auto xlm = makeNativeAsset();
+            auto idr = makeAsset(gateway, "IDR");
```

Comment:
Nitpick: `idr` and `usd` are unused.

Reply:
Thanks -- now I'm going to say that I have no idea how clang let me get away with that!  :)